### DISCUSSION TITLE
Reasons for using Pattern Lab

### DISCUSSION THREAD

**Discussion by backflip** [uswds/uswds#3969] [General]
Hi there!

I was wondering whether the reasons for [picking Pattern Lab](https://github.com/uswds/uswds/issues/3404) over other solutions like Storybook have been documented anywhere as they might help others finding themselves in the same situation.

Best regards,
Thomas

**Comment by mejiaj** [comment]
@backflip the more immediate need was to replace Fractal. Pattern lab has a twig option which is similar to nunjucks and we already had familiarity with it.

We did initial discovery here #3263. The initial problem at that time was the rework required to get other options working. We're working towards making the structure more modular so we can easily prototype and swap libraries in the future (if necessary).

**Reply by backflip** [reply]
@mejiaj, I see. Thanks a lot for elaborating!

**Reply by maya** [reply]
Storybook has a twig option with [storybook-twig](https://github.com/markhuot/storybook-twig). Storybook seems to be the top UI library tool today. @backflip @mejiaj

**Comment by drk4** [comment]
I am currently looking at possible options including Fractal, Pattern Lab and Storybook.
How did your migration to Pattern Lab go?

Also I was curious on what process you use to convert the pattern library into the documentation you have on https://designsystem.digital.gov/ as the formatting structure doesn't look like its taken from Fractal / Pattern Lab.

**Comment by thisisdano** [comment]
An update: We are changing course and moving our library to Storybook. More to come on this, but as we went through the process of migration, we learned more about what we really needed in a library tool. Storybook's modularity, structural flexibility, and relative lack of opinion became more and more valuable — and should prove to be increasingly valuable as we move forward with the design system. After some pain, we realized that Storybook was a better long-term solution for us, and it made sense to make the transition sooner rather than later.

**Reply by backflip** [reply]
@thisisdano, thanks a lot for the update, very much appreciated.

**Reply by drk4** [reply]
@thisisdano thanks for the update.
What Storybook framework have you decided to use out of interest?

**Reply by thisisdano** [reply]
We're using Storybook HTML with `twig-loader` and `twig.js` for Twig templates, but we're also experimenting with how we might support multiple template types. See in-progress work here: https://github.com/uswds/uswds/pull/4295

**Reply by maya** [reply]
Assuming you are also aware of [Chromatic](https://www.chromatic.com/), their visual regression testing Saas #3248. They also provide [open source sponsorships](https://www.chromatic.com/docs/open-source).

**Reply by thisisdano** [reply]
We're actually just getting started reviewing regression and unit testing improvements — thanks for the link!

**Reply by fafnirical** [reply]
Storybook! ❤️ Super excited to see how it goes. For unit/interaction testing, the Storybook team has been doing some interesting work to help enable [testing via stories](https://storybook.js.org/blog/ui-testing-playbook/) and [demoing interactions via Storybook itself](https://storybook.js.org/blog/interaction-testing-sneak-peek/).

**Reply by mejiaj** [reply]
Those demo interactions look amazing!

**Reply by maya** [reply]
When are you hoping to publish Storybook?
